lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1231,13 +1244,32 @@ class DifferentiableActivityInfo {
   Activity getActivity(SILInstruction *inst,
                        const SILAutoDiffIndices &indices) const;
 };
+
+class DifferentiableActivityCollection {
+public:
+  SmallDenseMap<GenericSignature *, DifferentiableActivityInfo> activityInfoMap;
+  SILFunction &function;
+  DominanceInfo *domInfo;
+  PostDominanceInfo *postDomInfo;
+
+  DifferentiableActivityInfo &getActivityInfo(GenericSignature *assocGenSig) {
+    DifferentiableActivityInfo info(*this, assocGenSig);
```

Comment:
A new info should not be created here before you check whether the map already has a value. It's extremely inefficient. `DifferentiableActivityInfo` constructor calls `analyze`.